File: frontend/src/entities/score.ts
import { Key } from "@/types/Key";
import { Entity, EntityRef } from "./entity";
import { Song } from "./song";
import { FileInfo } from "./fileInfo";

export interface Score extends Entity {
    song: EntityRef<Song>,
    description: string | undefined,
    fileUrl: string,
    key: Key | undefined,
    file?: FileInfo
}

File: backend/src/main/kotlin/nl/stevenbontenbal/chorister/domain/songs/Score.kt
package nl.stevenbontenbal.chorister.domain.songs

import jakarta.persistence.*
import nl.stevenbontenbal.chorister.domain.users.Choir

@Entity
class Score(
    override var id: Long? = null,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "CHOIR_ID")
    override var choir: Choir?,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "SONG_ID")
    override var song: Song,
    override var description: String?,
    override var key: Key?,
    var fileUrl: String?,
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "FILE_ID")
    var file: File?
) : Notation(id) {
    companion object
}